engine/lib/generic_printer/generic_printer.ml

```diff
@@ -477,51 +484,49 @@ module Make (F : Features.T) = struct
             self#expr'_GlobalVar_concrete ~super concrete
         | `Primitive primitive ->
             self#expr'_GlobalVar_primitive ~super primitive
+        | `TupleCons 0 -> string "tt"
         | _ ->
             self#assertion_failure
             @@ "GlobalVar: expected a concrete or primitive global ident, got:"
             ^ [%show: global_ident] global_ident
 
       method _do_not_override_pat'_PConstruct ~super ~constructor ~is_record
           ~is_struct ~fields =
-        match constructor with
-        | `Concrete constructor ->
-            let constructor =
-              self#_do_not_override_lazy_of_concrete_ident
-                AstPos_pat'_PConstruct_constructor constructor
-            in
-            let fields =
-              List.map
-                ~f:(fun field ->
-                  let { field; pat } = field#v in
-                  let field =
-                    match field with
-                    | `Concrete field -> field
-                    | _ ->
-                        self#assertion_failure
-                          "expr'.Construct: field: non-`Concrete"
-                  in
-                  let pat =
-                    self#_do_not_override_lazy_of_pat AstPos_field_pat__pat pat
-                  in
-                  ( self#_do_not_override_lazy_of_concrete_ident
-                      AstPos_pat'_PConstruct_fields field,
-                    pat ))
-                fields
-            in
-            self#pat'_PConstruct_inductive ~super ~constructor ~is_record
-              ~is_struct ~fields
-        | `TupleCons _ ->
-            let components =
-              List.map
-                ~f:(fun field ->
-                  self#_do_not_override_lazy_of_pat AstPos_field_pat__pat
-                    field#v.pat)
-                fields
-            in
-            self#pat'_PConstruct_tuple ~super ~components
-        | `Primitive _ | `TupleType _ | `TupleField _ | `Projector _ ->
-            self#assertion_failure "Construct unexpected constructors"
+        match (constructor, fields) with
+        | `TupleCons 0, [] -> string "'tt (* const unit *)" (* Unit pat *)
+        | `TupleCons 1, [ p ] -> p#p
```

Comment:
Same here, the Rust expression `(1u8,)` (a tuple of size 1) is different from `(1u8)` (a raw `u8`)
Also, unit should not be specific at the level of this file, that's a detail of implementation that should be decided by the backend, in `pat'_PConstruct_tuple`.
I don't understand what is wrong with the `_do_not_override_pat'_PConstruct` we have in main currently

Reply:
I should now call the correct function, instead of doing this hack. This was mostly done for debugging.